sotodlib/preprocess/processes.py

```diff
@@ -549,9 +550,21 @@ def gen_metric(cls, meta, proc_aman):
                 # mask those that were not set
                 nonzero = np.any(coeff != 0.0, axis=1)
 
+                # calculate amplitude of each mode
+                mode_labels = list(proc_aman.hwpss_stats.modes.vals)
+                num_re = re.compile("^[SC]([\d])$")
```

Comment:
Is this limited to a single digit number between 0-9? I think we could have >=10f.